src/map/ai/controllers/pet_controller.cpp

```diff
@@ -50,12 +50,18 @@ void CPetController::Tick(time_point tick)
 
 void CPetController::DoRoamTick(time_point tick)
 {
-    if ((PPet->PMaster == nullptr || PPet->PMaster->isDead()) && PPet->isAlive())
+    if ((PPet->PMaster == nullptr || PPet->PMaster->id == 0 || PPet->PMaster->name == "" || PPet->PMaster->isDead()) && PPet->isAlive() && PPet->objtype != TYPE_MOB)
```

Comment:
In what situations can we possibly get here where the id is 0 or the name is empty? This is a very very bad patch that sweeps a much bigger seeming problem under the rug? Can you explain when this happens, and why we aren't fixing it at the source?

Reply:
> In what situations can we possibly get here where the id is 0 or the name is empty? This is a very very bad patch that sweeps a much bigger seeming problem under the rug? Can you explain when this happens, and why we aren't fixing it at the source?

Yep after talking with shozo I have removed that part from the PR, thanks for pointing out!